### ISSUE TITLE
unstructured relationship mining algorithms: How do we scan texts, websites, blogs, links ....

### ISSUE THREAD

**Issue by Sandr0x00** [MusicConnectionMachine/RelationshipsG4#8]
How can we analyze blog entries, websites, texts (from links) for our keywords (defined in Relationships.txt, but with different languages, synonyms, ....). How do we get different languages? How do we get synonyms? Maybe there is a Service/Library we can use for analyzing texts effectively?

**Comment by simonzachau** [comment]
note from the seminar website: "http://orange.biolab.si can be used to explore the data sources, e.g. to extract relationships from unstructured data."

**Comment by sacdallago** [comment]
@simonzachau Orange is an exploratory tool with some neat and advanced functionalities wrapped around a fairly easy to understand UI. I would recommend using it on a subset of data to understand what the data looks like and which relationships to extract. **Maybe** it cannot manage the magnitude of data you will have, but it's just a supposition

**Comment by FelsyWaschbaer** [comment]
Here is a very nice overview on the topic https://web.stanford.edu/class/cs224u/materials/cs224u-2016-relation-extraction.pdf

**Comment by FelsyWaschbaer** [comment]
I guess we could use the [Stanford CoreNLP](http://stanfordnlp.github.io/CoreNLP/index.html) to do the relationship extraction. It has an openIE implementation and other usefull stuff. There is also a node.js wrapper for it (which I was not able to get to work so far)

**Comment by kordianbruck** [comment]
@simonzachau it would be nice to have a short comment, why this was closed as no commit/PR is referenced

**Comment by simonzachau** [comment]
@kordianbruck because we have finished [developing how to scan texts](https://github.com/MusicConnectionMachine/RelationshipsG4/commit/596586ffe285c94a7f63e0e4dab909b4b15f08f6) (This was the question of this issue). So the next step is to extract the relationships out of the triplets we have. [I opened a new issue](https://github.com/MusicConnectionMachine/RelationshipsG4/issues/25), which follows this one, and concentrates more on processing the json instead of scraping/formatting of the website content. 
I hope this explains my action.
Alternatively, I could reopen it (and optionally generalize the title).
